syntax = "proto2";

message TaskProto {
  enum Type {
       TryComplete = 1;
       ReorderSCC = 2;
       TryRunable = 3;
       TryRunableSCC = 4;
       TryCompleteSCC = 5;
       TryRun = 6;
  };
  required Type type = 1;
  repeated int64 data_ptr = 2;
  optional uint64 id = 3;
  optional uint64 other_id = 4;
  optional bool scc = 5;
  optional int64 vertex_ptr = 6;
  optional bool read = 7;
};

